src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save movie to DB: " + movie, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        try (SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+                Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(Movie.class, id));
```

Comment:
```suggestion
        try (Session session = sessionFactory.openSession()) {
            return Optional.ofNullable(session.get(Movie.class, id));
```
Если в try with resources напишешь sessionFactory, то твоя таблица будет дропаться каждый раз при вызове метода get()
<img width="690" alt="Screenshot 2021-12-20 at 15 46 31" src="https://user-images.githubusercontent.com/79812194/146901134-9ec3da22-ef53-45d9-8e69-a4539ff02d2b.png">